lib/sles4sap/sdaf_library.pm

```diff
@@ -0,0 +1,520 @@
+        "export SID=$args{sap_sid}",
+        "export ARM_SUBSCRIPTION_ID=$args{subscription_id}",
+        "export SAP_AUTOMATION_REPO_PATH=$deployment_dir/sap-automation/",
+        'export DEPLOYMENT_REPO_PATH=${SAP_AUTOMATION_REPO_PATH}',
+        "export CONFIG_REPO_PATH=$deployment_dir/WORKSPACES",
+        'export deployer_parameter_file=' . get_tfvars_path(deployment_type => 'deployer', vnet_code => $args{deployer_vnet_code}, %args),
+        'export library_parameter_file=' . get_tfvars_path(deployment_type => 'library', %args),
+        'export sap_system_parameter_file=' . get_tfvars_path(deployment_type => 'sap_system', vnet_code => $args{workload_vnet_code}, %args),
+        'export workload_zone_parameter_file=' . get_tfvars_path(deployment_type => 'workload_zone', vnet_code => $args{workload_vnet_code}, %args),
+        "export tfstate_storage_account=$args{sdaf_tfstate_storage_account}",
+        "export key_vault=$args{sdaf_key_vault}"
+    );
+
+    create_sdaf_os_var_file(\@variables);
+}
+
+=head2 get_tfvars_path
+
+    get_tfvars_path(
+        deployment_type=>$deployment_type,
+        env_code=>$env_code,
+        region_code=>$region_code,
+        [vnet_code=>$vnet_code,
+        sap_sid=>$sap_sid]);
+
+Returns full tfvars filepath respective to deployment type.
+
+B<deployment_type>: Type of the deployment (workload_zone, sap_system, library... etc)
+
+B<env_code>:  SDAF parameter for environment code (for our purpose we can use 'LAB')
+
+B<region_code>: SDAF parameter to choose PC region. Note SDAF is using internal abbreviations (SECE = swedencentral)
+
+B<vnet_code>: SDAF parameter for virtual network code. Library and deployer use different vnet than SUT env
+
+B<sap_sid>: SDAF parameter for sap system ID
+
+=cut
+
+sub get_tfvars_path {
+    my (%args) = @_;
+    my @supported_types = ('workload_zone', 'sap_system', 'library', 'deployer');
+    # common mandatory args
+
+    my @mandatory_args = qw(deployment_type env_code region_code);
+    # library does not require 'vnet_code'
+    push @mandatory_args, 'vnet_code' unless $args{deployment_type} eq 'library';
+    # only sap_system requires 'sap_sid'
+    push @mandatory_args, 'sap_sid' if $args{deployment_type} eq 'sap_system';
+
+    # Only workload and sap SUT needs unique ID
+    my $job_id = get_current_job_id();
+
+    foreach (@mandatory_args) { croak "Missing mandatory argument: '$_'" unless defined($args{$_}); }
+
+    croak "Invalid deployment type: $args{deployment_type}\nCurrently supported ones are: @supported_types" unless
+      grep(/^$args{deployment_type}$/, @supported_types);
+
+
+    my $file_path;
+    if ($args{deployment_type} eq 'workload_zone') {
+        my $env_reg_vnet = join('-', $args{env_code}, $args{region_code}, $args{vnet_code});
+        $file_path = "LANDSCAPE/$env_reg_vnet-INFRASTRUCTURE/$env_reg_vnet-INFRASTRUCTURE-$job_id.tfvars";
+    }
+    if ($args{deployment_type} eq 'deployer') {
```

Comment:
could be eventually converted to `elseif`

Reply:
Yep. good point. Switched to elsif.